lib/wasix/src/state/env.rs

```diff
@@ -580,61 +428,45 @@ impl WasiEnv {
     // FIXME: use custom error type
     #[allow(clippy::result_large_err)]
     pub(crate) fn instantiate(
-        mut init: WasiEnvInit,
+        self,
         module: Module,
-        module_hash: ModuleHash,
         store: &mut impl AsStoreMut,
-    ) -> Result<(Instance, WasiFunctionEnv), WasiRuntimeError> {
-        let call_initialize = init.call_initialize;
-        let spawn_type = init.memory_ty.take();
-
-        if init.extra_tracing {
-            for import in module.imports() {
-                tracing::trace!("import {}.{}", import.module(), import.name());
-            }
-        }
-
-        let additional_imports = init.additional_imports.clone();
+        memory: Option<Memory>,
+        update_layout: bool,
+        call_initialize: bool,
+    ) -> Result<(Instance, WasiFunctionEnv), WasiThreadError> {
+        let is_dl = super::linker::is_dynamically_linked(&module);
 
-        let env = Self::from_init(init, module_hash)?;
-        let pid = env.process.pid();
+        let pid = self.process.pid();
 
         let mut store = store.as_store_mut();
 
-        let tasks = env.runtime.task_manager().clone();
-        let mut func_env = WasiFunctionEnv::new(&mut store, env);
-
-        // Determine if shared memory needs to be created and imported
-        let shared_memory = module.imports().memories().next().map(|a| *a.ty());
-
```

Comment:
I'm positive that this was there for a reason, because some WASI (not WASIX) modules , depending on how they are built, sometimes either export or import the memory, so this was here to make sure things always work.

Reply:
The functionality still exists somewhere around TaskWasm. There was 2 of it, now there's just one.